content/en/agent/faq/linux-agent-2022-key-rotation.md

```diff
@@ -0,0 +1,124 @@
+</div>
+
+If you're using one of the following install methods, your hosts trust the key automatically and no further action is needed:
+
+* [Agent install script][5] v1.5.0 or later (released Jun 29, 2021)
+* [Chef cookbook][6] v4.11.0 or later (released Aug 10, 2021)
+* [Ansible role][7] v4.9.0 or later (released May 6, 2021)
+* [Puppet module][8] v3.13 or later (released Aug 11, 2021)
+* [SaltStack formula][9] v3.4 or later (released Aug 12, 2021)
+* [Elastic Beanstalk][10] config templates updated as of Mar 29, 2021 (should contain `DATADOG_RPM_KEY_FD4BF915.public` under `gpgkey`)
+* Containerized Agents (Docker/Kubernetes): No action needed on any version
+* Heroku: No action needed on any version.
+* Windows/MacOS Agents: No action needed on any version.
+
+Additionally, if you are using the DEB version of the Agent version 7.31.0 or greater, your hosts should have installed the `datadog-signing-keys` package that automatically trusts the new key.
+
+If unsure, read the steps in [Check if a host trusts the new GPG key](#check-if-a-host-trusts-the-new-gpg-key).
+
+Trying to install or upgrade the Agent package without trusting the new key results in `NOKEY` errors when installing the RPM package and `NO_PUBKEY` errors when installing the DEB package.
+
+For hosts running older versions of the install methods listed above or older versions of the DEB package, Datadog recommends updating the install method to the latest version. Alternatively, Debian and Ubuntu users can update the Agent to version 7.31.0 or greater. Otherwise, the keys can be manually updated on host by following these instructions:
+
+### Manual update
+
+{{< tabs >}}
+{{% tab "Debian/Ubuntu" %}}
+
+Run the following commands on the host:
+
+```bash
+$ wget -qO - https://keys.datadoghq.com/DATADOG_APT_KEY_F14F620E.public | sudo apt-key add -
+```
+
+{{% /tab %}}
+{{% tab "RedHat/CentOS" %}}
+
+Edit the `/etc/yum.repos.d/datadog.repo` and make sure it includes the following list of keys in the `gpgkey` section:
+
+```
+gpgkey=https://keys.datadoghq.com/DATADOG_RPM_KEY_CURRENT.public
+       https://keys.datadoghq.com/DATADOG_RPM_KEY_FD4BF915.public
+       https://keys.datadoghq.com/DATADOG_RPM_KEY_E09422B3.public
+       https://keys.datadoghq.com/DATADOG_RPM_KEY.public
+```
+
+{{% /tab %}}
+{{% tab "SUSE" %}}
+
+Edit the `/etc/zypp/repos.d/datadog.repo` and make sure it includes the following list of keys in the `gpgkey` section:
+
+```
+gpgkey=https://keys.datadoghq.com/DATADOG_RPM_KEY_CURRENT.public
+       https://keys.datadoghq.com/DATADOG_RPM_KEY_FD4BF915.public
+       https://keys.datadoghq.com/DATADOG_RPM_KEY_E09422B3.public
+       https://keys.datadoghq.com/DATADOG_RPM_KEY.public
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+## Check if a host trusts the new GPG key
+
+
+{{< tabs >}}
+{{% tab "Debian/Ubuntu" %}}
+
+Run the following commands on the host:
+
+```bash
+$ apt-key list
```

Comment:
If we keep doing `apt-key add` above, then this is true; otherwise if we advise to use the separate keyring, this will need to be `gpg --no-default-keyring --keyring /usr/share/keyrings/datadog-archive-keyring.gpg --list-keys`.

Reply:
This should actually ask you to run both commands, because you could have the key added either way.

Reply:
Thinking about this further, I *think* that all cfg management tools versions that create the separate keyring also import the new key into it, but you're right that if people want to verify, they need to be provided both versions of what to do based on how they set the system up.

Reply:
I've left it just as `rpm -q` on rpm-based and `apt-key list` on debian-based. Once I test it I'll check if it works on all distros (it will only work if we always add the key to the "generic" keyring instead of adding it per-repo).